lib/registries/addon/discover/controller.ts

```diff
@@ -224,95 +245,67 @@ export default class Discover extends Controller.extend(discoverQueryParams.Mixi
-        }
-
-        for (const type of event.queryParams.registrationTypes) {
-            filters.push(new ShareTermsFilter('registration_type', type, type));
-        }
-
-        const order = this.sortOptions.find(option =>
-            !!option.key
-            && event.queryParams.sort.endsWith(option.key)
-            && option.ascending === !event.queryParams.sort.startsWith('-'));
-
-        this.get('doSearch').perform(new SearchOptions({
-            query: event.queryParams.query,
-            size: event.queryParams.size,
-            page: event.queryParams.page,
-            order: order || this.sortOptions[0],
-            filters: OrderedSet(filters),
-        }), { noPageReset: true });
-    }
-
-    setQueryParams(this: Discover, options: SearchOptions) {
-        this.set('page', options.page);
-        this.set('size', options.size);
-        this.set('query', options.query || '');
-
-        // date_modified is a weird case, so don't save it in a query param.
-        if (options.order.key !== 'date_modified') {
-            this.set('sort', `${options.order.ascending ? '' : '-'}${options.order.key || ''}`);
-        }
-
-        const providers: string[] = [];
-        const registrationTypes: string[] = [];
+    @action
+    onSearchOptionsUpdated(this: Discover, options: SearchOptions) {
+        const sources: ShareTermsFilter[] = [];
+        const registrationTypes: ShareTermsFilter[] = [];
         for (const filter of options.filters.values()) {
             if (filter.key === 'sources') {
-                providers.push(filter.value as string);
+                sources.push(filter as ShareTermsFilter);
             }
 
             if (filter.key === 'registration_type') {
-                registrationTypes.push(filter.value as string);
+                registrationTypes.push(filter as ShareTermsFilter);
             }
         }
 
-        this.set('sources', providers);
-        this.set('registrationTypes', registrationTypes);
-    }
+        const changes = {} as any;
 
-    @action
-    onSearchOptionsUpdated(this: Discover, options: SearchOptions) {
-        this.get('doSearch').perform(options);
+        if (!isEqual(this.sources, sources)) {
+            changes.page = 1;
+            changes.sources = sources;
+        }
+
+        if (!isEqual(this.registrationTypes, registrationTypes)) {
+            changes.page = 1;
+            changes.registrationTypes = registrationTypes;
+        }
+
+        // If any filters are changed page is reset to 1
+        this.setProperties(changes);
     }
 
     @action
     changePage(this: Discover, page: number) {
-        this.get('doSearch').perform(
-            this.searchOptions.set('page', page),
-            { scrollTop: true },
+        this.set('page', page);
+
+        // Scroll to the top of results on pagination
+        $('html, body').scrollTop(
+            $(`.${discoverStyles.Discover__Body}`).parent().position().top,
```

Comment:
jquery grumble grumble

Reply:
I think I actually know how to fix that!